src/app/program-details/program-details.page.ts

```diff
@@ -0,0 +1,159 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { NavController } from '@ionic/angular';
+import { ToastService } from '../services/toast/toast.service';
+import { LoaderService } from '../services/loader/loader.service';
+import { ProjectsApiService } from '../services/projects-api/projects-api.service';
+import { finalize } from 'rxjs';
+import { UrlConfig } from 'src/app/interfaces/main.interface';
+import urlConfig from 'src/app/config/url.config.json';
+import { actions } from '../config/actionContants';
+import { ProfileService } from '../services/profile/profile.service';
+
+@Component({
+  selector: 'app-program-details',
+  templateUrl: './program-details.page.html',
+  styleUrls: ['./program-details.page.scss'],
+})
+export class ProgramDetailsPage implements OnInit {
+  programId:any;
+  showLoading:boolean = true;
+  characterLimit = 150;
+  showMore:boolean=false
+  programFilters=actions.PROGRAM_FILTERS;
+  programFilter:any = "all";
+  programList:any=[];
+  entityData:any;
+  programData:any;
+  filteredList: any;
+  filterData:any;
+ 
+  constructor(private route : ActivatedRoute,
+    private navCtrl: NavController,
+    private toastService :ToastService, 
+    private loader : LoaderService,
+    private ProjectsApiService: ProjectsApiService,
+    private router: Router,
+    private profileService: ProfileService) {
+
+    this.route.params.subscribe(param=>{
+      this.programId = param['id'];
+    })
+
+ }
+
+  ngOnInit() {
+  }
+
+  ionViewWillEnter(){
+    this.getProfileDetails()
+  }
+  ionViewWillLeave(){
+    this.programList=[]
+    this.filteredList=[]
+    this.filterData = []
+  }
+  goBack() {
+    this.navCtrl.back();
+  }
+
+  filterChanged(event:any){
+    this.programFilter = event.detail.value;
+    this.applyFilter();
+  }
+
+  toggleReadMore() {
+    this.showMore = !this.showMore;
+  }
+
+  selectSection(data:any){
+
+    this.filterData.forEach((element:any) => {
+      if(data.sectionName == element.sectionName){
+        data.show = ! data.show;
+      }else{
+        element.show =false;
+      }
+    });
+
+  }
+
+  formatList(){
+    this.filterData=this.programData.data;
+    this.filterData.forEach((data:any)=>{
+      let sectionName=data.type=='improvementProject'?'projects':data.type+'s'
+      let index = this.programList.findIndex((val:any)=>{return val.sectionName==sectionName})
+      if(index!==-1){
+        this.programList[index].sectionList.push(data)
+       }else{
+        let order=data.type=='improvementProject'?0:data.type=='observation'?1:2
+        this.programList.push({sectionName:sectionName,sectionList:[data],order:order})
+       }
+    })
+    this.filterData=this.programList.sort((a:any,b:any)=>{return a.order - b.order}).map((item:any)=>({ ...item,show:true}))
+  }
+
+  get descriptionText(): string {
```

Comment:
Without using get method try to do the formatting and store it in a variable and use that variable in the html